services/src/datasets/postgres.rs

```diff
@@ -360,30 +357,30 @@ where
     <Tls as MakeTlsConnect<Socket>>::TlsConnect: Send,
     <<Tls as MakeTlsConnect<Socket>>::TlsConnect as TlsConnect<Socket>>::Future: Send,
 {
-    fn to_json(&self) -> Result<DatasetMetaDataJson> {
+    fn to_typed_metadata(&self) -> Result<DatasetMetaData> {
```

Comment:
do we still need this whole ordeal (meaning the PostgresStorable trait) if we no longer have the requirement that something is json serializable?

Reply:
I wanted to get rid of it, but unfortunately, the API only offers you some Trait type that isn't the enum.
Thus, I still need the function to get something that implements FromSql/ToSql.